### PR TITLE
Add settings max_unexpected_write_error_retries for Azure Blob Storage

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added setting for Azure Blob Storage `azure_max_unexpected_write_error_retries` , can also be set from config under azure section.

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit d22fc3a224ac29857f3dc3eb60ff872221829006 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59001/d22fc3a224ac29857f3dc3eb60ff872221829006/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59001/d22fc3a224ac29857f3dc3eb60ff872221829006/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Core/Settings.h [hunk 1]
```diff
@@ -86,6 +86,7 @@ class IColumn;
     M(UInt64, azure_max_single_part_upload_size, 100*1024*1024, "The maximum size of object to upload using singlepart upload to Azure blob storage.", 0) \
     M(UInt64, s3_max_single_read_retries, 4, "The maximum number of retries during single S3 read.", 0) \
     M(UInt64, azure_max_single_read_retries, 4, "The maximum number of retries during single Azure blob storage read.", 0) \
+    M(UInt64, azure_max_unexpected_write_error_retries, 4, "The maximum number of retries in case of unexpected errors during Azure blob storage write", 0) \
```
Comment (kssenii) [inline src/Core/Settings.h:89]: may be just "azure_max_single_write_retries" for consistency with other such settings?

Comment (SmitaRKulkarni) [reply src/Core/Settings.h:89]: I kept is same as '`s3_max_unexpected_write_error_retries`' which I think is used in similar places. We use this for write exception in multipart upload, so not sure if its would be good to call it `single_write_retries`

Comment (kssenii) [reply src/Core/Settings.h:89]: Ah, did not see it before, now I see, indeed it is better then.

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
> AST fuzzer (ubsan)

https://github.com/ClickHouse/ClickHouse/issues/57566

> Stateless tests (release, analyzer) — fail: 2

unrelated to changes

**Comment by Algunenano** [comment]
This has broken master: https://s3.amazonaws.com/clickhouse-test-reports/59329/ab2b12ba9f9a33e185362244eb396f0d111865c8/upgrade_check__debug_.html

**Pull request merged**